l1-contracts/contracts/bridge/L1AssetRouter.sol

```diff
@@ -412,59 +199,144 @@ contract L1AssetRouter is IL1AssetRouter, ReentrancyGuard, Ownable2StepUpgradeab
             _prevMsgSender: _prevMsgSender,
             _assetId: assetId,
             _l2BridgeMintCalldata: l2BridgeMintCalldata,
-            _txDataHash: txDataHash
+            _txDataHash: txDataHash,
+            _deposit: true
         });
 
         emit BridgehubDepositInitiated({
             chainId: _chainId,
             txDataHash: txDataHash,
             from: _prevMsgSender,
             assetId: assetId,
-            l2BridgeMintCalldata: l2BridgeMintCalldata
+            bridgeMintCalldata: l2BridgeMintCalldata
         });
     }
 
-    /// @notice Forwards the burn request for specific asset to respective asset handler.
-    /// @param _chainId The chain ID of the ZK chain to which deposit.
-    /// @param _l2Value The L2 `msg.value` from the L1 -> L2 deposit transaction.
-    /// @param _assetId The deposited asset ID.
+    /// @notice Initiates a withdrawal by burning funds on the contract and sending the message to L1
+    /// where tokens would be unlocked.
+    /// @param _chainId The chain ID of the ZK chain to which to withdraw.
     /// @param _prevMsgSender The `msg.sender` address from the external call that initiated current one.
-    /// @param _transferData The encoded data, which is used by the asset handler to determine L2 recipient and amount. Might include extra information.
-    /// @return l2BridgeMintCalldata The calldata used by remote asset handler to mint tokens for recipient.
-    function _burn(
+    /// @param _assetId The encoding of the asset on L2 which is withdrawn.
+    /// @param _transferData The data that is passed to the asset handler contract.
+    function bridgehubWithdraw(
+        // ToDo: how to unify L1 and L2 withdraw? 1 is done via system contract, another one via request, so input / output is inhenerently different
```

Comment:
DM'ed this question

Reply:
Will add it on L2 as well

Reply:
The bridgehubTransfer/Deposit and bridgehubWithdraw should be unified into one function, only the NTV should care if the token is native or not